proposals/Reliable Mutability Primitive.md

```diff
@@ -0,0 +1,166 @@
+# Reliable Mutability Primitive
+
+Authors: [@gozala](https://github.com/gozala)
+
+Initial PR: https://github.com/protocol/web3-dev-team/pull/19/
+<!--
+This template is intended to be used by those who would like to pitch a new project for one of the Web3 Dev project teams to take on. It should contain sufficient detail that others can understand how this project contributes to our team’s mission of  product-market fit for our unified stack of protocols, what is included in scope of the project, where to get started if a project team were to take this on, and any other information relevant for prioritizing this project against others.
+Good project scope aims for ~3-5 engineers for 1-3 months (though feel free to suggest larger-scoped projects anyway). Projects do not include regular day-to-day maintenance and improvement work, e.g. on testing, tooling, validation, code clarity, refactors for future capability, etc.
+-->
+
+### Purpose &amp; impact 
+##### Background &amp; intent
+_Describe the desired state of the world after this project? Why does that matter?_
+<!--
+Outline the status quo, including any relevant context on the problem you’re seeing that this project should solve. Wherever possible, include pains or problems that you’ve seen users experience to help motivate why solving this problem works towards top-line objectives. 
+-->
+
+
+Today ecosystem has to work around the lack of functional mutability primitives. [IPNS][] does not provide reliable solution forcing community to adopt different custom solutions.
+
+- [Pinata][] gets asked about how to change content for CID once every few weeks _(Most of the time they find that is not what the users actually need)_.
+- [fission.codes](https://fission.codes/) found IPNS persistance unreliable and had to build custom solution that updates [DNSLink][] records (**TODO**: Get a quote / link from Boris)
+- **TODO**: Survey textile, last time we talked they found IPNS to be unreliable for propagating thread updates.
```

Comment:
[@carsonfarmer: This is still correct, IPNS is too unreliable for anything with essentially sub-minute temporal resolution. So for data updates, it is not useful. However, we are finding it quite useful for buckets, and have used it to great effect recently.](https://hackmd.io/@gozala/ipns-pitch)

Reply:
I should also mention something else to provide further context: We're currently managing 1000s (yes, that's right) of IPNS updates for buckets. This currently takes an extremely long time to republish. We are actively conducting tests to determine exactly how long this takes, and how often it fails to complete. In some cases, it is taking hours per day.